File: compiler/listeners/text.py
#!/usr/bin/python3
from generated.ProzeParser import ProzeParser
from generated.ProzeListener import ProzeListener


class TextListener(ProzeListener):

    def __init__(self):
        self._output = ''
    
    def get_output(self):
        return self._output
    
    def enterTitle_tag(self, ctx: ProzeParser.Title_tagContext):
        super().enterTitle_tag(ctx)
        result = []
        for word in ctx.markup_value().WORD():
            result.append(word.getText())
        self._output += ' '.join(result) + '\n'

File: compiler-typescript/src/listeners/text.ts
import {Title_tagContext} from '../../generated/ProzeParser';
import {ListenerOutput} from './interface';


export class TextListener implements ListenerOutput {

    private output: string;

    constructor() {
        this.output = '';
    }

    getOutput(): string {
        return this.output;
    }

    enterTitle_tag(ctx: Title_tagContext): void {
        let result: string[] = [];
        for (let word of ctx.markup_value().WORD()) {
            if (word.payload.text != undefined) {
                result.push(word.payload.text);
            }
        }
        this.output += result.join(' ') + '\n';
    }

}